prime-router/docs/features/0005-managing-translation-schemas.md

```diff
@@ -0,0 +1,765 @@
+# Managing Translation Schemas
+
+## Context
+
+Translation schemas are used at multiple points in the universal pipeline
+
+- transforming items sent by senders
+- transforming items right before they are delivered to a receiver
+
+and cover different kinds of transforms:
+
+- HL7 -> FHIR
+    - **This is handled by another [library](https://github.com/LinuxForHealth/hl7v2-fhir-converter) with limited configuration option of where to read the mapping files from**
+- FHIR -> FHIR
+- FHIR -> HL7
+
+These translation schemas are currently stored as files that are included in the deployed application and read from the file sytem
+when they are applied to an item.
+
+## Problem
+
+The overarching problem is that since the translation schemas are part of the deployed application they can only be updated at the frequency at which the application itself is deployed.  This is quite limiting and has downstream impacts such as:
+
+- onboarding new senders/receivers can take a while as iterative changes to the translation schema have to be tested out over several days or weeks
+- bugs discovered in the translation schema cannot be immediately addressed without a hotfix
+- the senders/receivers do not have any capability to self-serve changes to their own translation schemas
+
+## Goal
+
+Design a new implementation that enables iterating on the translation schemas outside of the deployments and unlocks the ability to implement a feature that enables sender/receivers to self-serve on changes.
```

Comment:
Again, this goal should be reworded a bit since users self serving isn't a goal of ours